### DISCUSSION TITLE
COTS UE can not see 5G network

### DISCUSSION THREAD

**Discussion by peihaoY** [srsran/srsRAN_Project#484] [General Help]
I am following this page(https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html) to establish a connection between UE and gNB, but unlike the tutorial, I am using 5G Docker. I can successfully connect the gNB and srsUE, and they work well. However, I am unable to see the 5G network with my Xiaomi 11 Lite 5G. I have attached my configuration as follows.
UE：
Xiaomi 11 Lite 5G
GNB and 5GCore：
USRP N310, Ubuntu 22.04, UHD 4.5.0
SIM card:
sysmoISIM-SJA5
External clock:
GPSDO 
[gnb and 5gcore.zip](https://github.com/srsran/srsRAN_Project/files/14351783/gnb.and.5gcore.zip)
I have used the SA network, and my two SIM cards do not have services 124 and 125. The PLMNs are 00101 and 99970, band n3 and n78, respectively, and neither of them works. Thank you for your help.

**Comment by herlesupreeth** [comment]
Try typing `*#*#726633#*#*` on the dial pad. You should see a hidden menu, there enable **Mode SA**

**Reply by peihaoY** [reply]
Actually, I have enabled Mode SA. It cannot see the 5G network.

**Reply by herlesupreeth** [reply]
hmmm.. have you tried adjusting the Tx and Rx gain values?

**Reply by peihaoY** [reply]
Yes. I have tried the maximum gain Tx Rx = 65.

**Reply by Thanasislt** [reply]
Hello, I can confirm this behaviour as well on our testbed. 
We deployed Open5GS (SA, v2.6.0) with srsRAN 23.10 (Dec 2023) and a USRP X310 as gnb. 
Xiaomi 11 Lite 5G could not connect despite our attempts.

We ended up rooting the device and installing PixelOS. Afterwards, the device connects to the 5G network with no problems.

**Comment by brendan-mcauliffe** [comment]
Have you forced 5G using an app like shown in the tutorial?

**Reply by peihaoY** [reply]
Yes. After that, I occasionally see T-mobile 5G, but most of the time there's nothing..

**Comment by pgawlowicz** [comment]
do you run dockerized version of the Open5GS? could you post your gnb logs?

**Comment by peihaoY** [comment]
Yes, I used the dockerized version. The gnb log is as follows:
[gnb2.log](https://github.com/srsran/srsRAN_Project/files/14407618/gnb2.log)
When the following occurs  "Late: 0; Underflow: 2; Overflow: 0;" the log file is as follows:
[gnb1.log](https://github.com/srsran/srsRAN_Project/files/14407643/gnb1.log)

**Reply by robertfalkenberg** [reply]
According to these logs it seems there has been a successful attach and registration, the UE got the IP 10.45.1.3 from the core:

```
open5gs_5gc  | 02/28 23:49:35.467: [smf] INFO: UE SUPI[imsi-001010123456789] DNN[srsapn] IPv4[10.45.1.3] IPv6[] (../src/smf/npcf-handler.c:528)
open5gs_5gc  | 02/28 23:49:35.467: [upf] INFO: [Added] Number of UPF-Sessions is now 1 (../src/upf/context.c:194)
open5gs_5gc  | 02/28 23:49:35.467: [gtp] INFO: gtp_connect() [127.0.0.4]:2152 (../lib/gtp/path.c:60)
open5gs_5gc  | 02/28 23:49:35.467: [upf] INFO: UE F-SEID[UP:0x1 CP:0x1] APN[srsapn] PDN-Type[1] IPv4[10.45.1.3] IPv6[] (../src/upf/context.c:467)
open5gs_5gc  | 02/28 23:49:35.467: [upf] INFO: UE F-SEID[UP:0x1 CP:0x1] APN[srsapn] PDN-Type[1] IPv4[10.45.1.3] IPv6[] (../src/upf/context.c:467)
```

Moreover, a data radio bearer (DRB) was also exchanging some data, as you can see in the gNB logs:
```
...
2024-02-28T23:49:55.819504 [RLC     ] [I] du=1 ue=1 DRB1 UL: RX PDU. pdu_len=66 dc=data p=1 si=full sn=19
2024-02-28T23:49:55.819505 [RLC     ] [I] du=1 ue=1 DRB1 UL: RX SDU. sn=19 sdu_len=63
2024-02-28T23:49:55.819505 [PDCP    ] [I] ue=0 DRB1 UL: RX PDU. type=data pdu_len=63 sn=19 count=19
2024-02-28T23:49:55.819506 [PDCP    ] [I] ue=0 DRB1 UL: RX SDU. count=19
2024-02-28T23:49:55.819508 [GTPU    ] [I] ue=0 UL teid=0x000002: TX PDU. pdu_len=76 teid=0x000002 qfi=QFI=1
...
```

**Reply by peihaoY** [reply]
Yes, I think so. I can see some brate on gNB with some overflow. But why my phone always shows "cannot connect to the network"?
```
Late: 0; Underflow: 6; Overflow: 0;
           -----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4612   10   1   10   1.1k    6    0   0%      0 |  10.3   12    25k    6    0   0%      0
   1 4612    9   1   10    912    5    0   0%      0 |  10.5   12    21k    5    0   0%      0
Late: 0; Underflow: 5; Overflow: 0;
   1 4612   10   1   11    936    5    0   0%      0 |  10.1   12    21k    5    0   0%      0
Late: 74; Underflow: 5; Overflow: 0;
```

**Reply by robertfalkenberg** [reply]
```
Late: 74; Underflow: 5; Overflow: 0;
```
74 is a lot... you could reduce log level to `warning`. Maybe this relaxes the stress on the computer a bit.

**Comment by pgawlowicz** [comment]
hmm, a lot of RF issues. Please try to run this script: https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance
to set the CPU in the performance mode.

**Reply by peihaoY** [reply]
Actually, I always run this file.

**Reply by pgawlowicz** [reply]
Do you have another phone to try? People reported already that they cannot connect some of the Xiaomi phones to srsgnb.

**Reply by peihaoY** [reply]
We just placed an order for the oneplus nord 5G. I will try and then feedback.

**Reply by peihaoY** [reply]
I tried the OnePlus phone and indeed, it was very easy to see the 5G network(only n78, cannot see anything when using n3). Thanks for your information. The 5G core Docker shows the following: [added]  Number of SMF-UEs is now 1, so I think they connected. However, the phone displays "forbidden" or cannot connect. Could you please help me look at the log files again? I can't figure out the problem. Below I have attached my log files.
[5gcore docker.txt](https://github.com/srsran/srsRAN_Project/files/14442440/5gcore.docker.txt)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/14442437/gnb.log)

**Reply by pgawlowicz** [reply]
Did you fill the Open5gs subscriber database correctly? It seems you only have the default imsi : `001010123456789`. I guess it is not the one of the used simcard?
in the open5gs logs there is `Authentication failure(Synch failure)` and then the core sends `UE Context Release` command.

**Reply by peihaoY** [reply]
I filled the Open5gs subscriber database:
`SUBSCRIBER_DB=001010123456789,41B7157E3337F0ADD8DA89210D89E17F,opc,1CD638FC96E02EBD35AA0D41EB6F812F,8000,9,10.45.1.3`
If I use a SIM card that is not in the Open5GS subscriber database, it displays as follows:
`open5gs_5gc  | 02/28 23:13:09.787: [dbi] INFO: [imsi-001010000105012] Cannot find IMSI in DB (../lib/dbi/subscription.c:56)`

**Reply by pgawlowicz** [reply]
But does your simcard have IMSI=001010123456789?

**Reply by peihaoY** [reply]
Yes, it has. This is my sim card information.
![微信图片_20240229111213](https://github.com/srsran/srsRAN_Project/assets/147773058/7aece099-597b-4e17-9ab0-9c9a4efde81a)

**Reply by peihaoY** [reply]
But it can't connect. Do you have any other comments on this problem?
![微信图片_20240229113742](https://github.com/srsran/srsRAN_Project/assets/147773058/02625881-8ba8-4cb7-9f9a-2947af035e4a)![微信图片_20240229113746](https://github.com/srsran/srsRAN_Project/assets/147773058/d1e67bf1-bb42-46ae-a835-62a93fa07b35)

**Reply by pgawlowicz** [reply]
1. Did you already tried [this](https://github.com/srsran/srsRAN_Project/discussions/135#discussioncomment-6521094) ?
2. Change PLMN to 90170. The [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html) says:

_The PLMN is changed to 90170. Using the OnePlus 8t, we have found that by forcing a roaming scenario, the phone sees the network and attaches more reliably. A roaming scenario is forced by setting different PLMNs in the cell and the ISIM in the phone._

**Reply by peihaoY** [reply]
1. Neither of these methods works for me; the display remains the same.
2. When I use PLMN = 90170, the phone displays 'forbidden', and there's no output on the 5G core and gNB console. When I use 00101, I can get the logs mentioned above and see some brate and many overflows on the gNB, but the phone displays 'cannot connect to the network.' It shows as in the following picture when automatically connecting, but I still can't access the internet. 
![微信图片_20240301111209](https://github.com/srsran/srsRAN_Project/assets/147773058/f93bbb07-e504-461a-9e57-8dbfed959368)
I wonder if this warning has anything to do with it?"
`open5gs_5gc  | 02/28 23:49:24.579: [gmm] WARNING: Authentication failure(Synch failure) (../src/amf/gmm-sm.c:1366)`
`open5gs_5gc  | 02/28 23:49:35.467: [smf] WARNING: Unknown PCO ID:(0x23) (../src/smf/context.c:3035)`
`open5gs_5gc  | 02/28 23:49:35.467: [smf] WARNING: Unknown PCO ID:(0x24) (../src/smf/context.c:3035)`
Or is there an issue with this mac.pcap?[gnb_mac.zip](https://github.com/srsran/srsRAN_Project/files/14463382/gnb_mac.zip)

**Reply by robertfalkenberg** [reply]
I wouldn't worry too much about this message. I also get those for quite a long time; the UE attaches eventually.

```
open5gs_5gc  | 03/01 14:33:20.485: [gmm] INFO: [suci-0-001-01-0-0-0-0000051520]    SUCI (../src/amf/gmm-handler.c:880)
open5gs_5gc  | 03/01 14:33:20.565: [gmm] WARNING: Authentication failure(Synch failure) (../src/amf/gmm-sm.c:1366)
open5gs_5gc  | 03/01 14:33:20.846: [gmm] INFO: [imsi-001010000051520] Registration complete (../src/amf/gmm-sm.c:1916)
open5gs_5gc  | 03/01 14:33:20.846: [amf] INFO: [imsi-001010000051520] Configuration update command (../src/amf/nas-path.c:612)

...
open5gs_5gc  | 03/01 14:33:31.042: [upf] INFO: UE F-SEID[UP:0x1 CP:0x1] APN[srsapn] PDN-Type[1] IPv4[10.45.1.121] IPv6[] (../src/upf/context.c:467)
open5gs_5gc  | 03/01 14:33:31.042: [upf] INFO: UE F-SEID[UP:0x1 CP:0x1] APN[srsapn] PDN-Type[1] IPv4[10.45.1.121] IPv6[] (../src/upf/context.c:467)
open5gs_5gc  | 03/01 14:33:31.042: [gtp] INFO: gtp_connect() [10.53.1.2]:2152 (../lib/gtp/path.c:60)
open5gs_5gc  | 03/01 14:33:31.043: [smf] WARNING: Unknown PCO ID:(0x23) (../src/smf/context.c:3035)
open5gs_5gc  | 03/01 14:33:31.043: [smf] WARNING: Unknown PCO ID:(0x24) (../src/smf/context.c:3035)
open5gs_5gc  | 03/01 14:33:31.126: [gtp] INFO: gtp_connect() [10.53.1.1]:2152 (../lib/gtp/path.c:60)

```

**Reply by robertfalkenberg** [reply]
Regarding your MAC PCAP, I can see the UE had an active DRB from
```
41	2024-02-29 19:59:32,208591	10.45.1.3	8.8.8.8	DNS	543			UEId=1     [UL] [AM] DRB:1  [DATA]     SN=0               [63-bytes]   ||   UEId=1     [UL] [AM] DRB:1  [DATA]     SN=1               [78-bytes]   ||   UEId=1     [UL] [AM] DRB:1  [DATA] (P) SN=2               [65-bytes]  (Short BSR LCG ID=0 BS=0)   (PHR PH=51 PCMAX_f_c=51)   (Padding 285 bytes) 
```
to
```
375	2024-02-29 20:01:54,129615			RLC-NR	100			UEId=1     [DL] [AM] DRB:1  [CONTROL]  ACK_SN=248     (Padding 63 bytes) 
```

From 19:59:32 .. 20:01:54 are a few minutes before the UE jumps off.
Could is be this? https://github.com/srsran/srsRAN_Project/discussions/100

**Reply by peihaoY** [reply]
Great! Now it can stay attached without any overflow. They can ping each other. Thanks!

**Comment by peihaoY** [comment]
I'm having another problem with my oneplus. Why does it disconnect whenever it tries to test the upload speed or transfer more data via iperf? After a while it may connect again automatically, mostly it doesn't. Then the 5g core console comes up with a lot of warnings as shown below. Attached are my log files. 
[gnb_log+mac+ngap and 5gcore_log.zip](https://github.com/srsran/srsRAN_Project/files/14548816/gnb_log%2Bmac%2Bngap.and.5gcore_log.zip)
![onepluse_mistake](https://github.com/srsran/srsRAN_Project/assets/147773058/26b9f41a-d659-452d-8290-d5cb58811b47)
![Image20240309224417](https://github.com/srsran/srsRAN_Project/assets/147773058/b794499c-92c5-4433-ab58-909ede10227c)

**Reply by pgawlowicz** [reply]
UL SNR goes down and probably the UE is released due to Radio Link Failure (RLF) event. Coudl you reduce the RX-gain at gnb and try again?

**Reply by peihaoY** [reply]
I reduced the rx-gain to 25, 35, 45 to test separately and changed the value of tx-gain as well. My problem is not solved. Are there any other ways I can try to solve the problem?

**Reply by pgawlowicz** [reply]
could you provide your gnb and srsue configs and logs?

**Reply by peihaoY** [reply]
I used onplus as UE. Attached is gnb config and log.
[gnb config and log.zip](https://github.com/srsran/srsRAN_Project/files/14562249/gnb.config.and.log.zip)

**Reply by pgawlowicz** [reply]
did you try with external clock?

**Reply by peihaoY** [reply]
Yes, I used an external clock, but the same issue still occurred. My Motorola phone has never had issues with disconnection. So, I'm wondering if it might be an issue with the OnePlus phone? However, you have already tested that the connection with this phone is fine. Do you have any further comments?

**Reply by pgawlowicz** [reply]
could you install NetMonster app (or similar) and check what is the RSRP and SNR in DL?

**Reply by peihaoY** [reply]
NetMonster shows `Not even a single cell`. Another smartphone can see `1 01 5G.SA 3500`.

**Reply by pgawlowicz** [reply]
@peihaoY  Any updates on this issue?
If the UE connects to the network and is able to ping, the Netmonster app should show the info about the cell.

**Reply by peihaoY** [reply]
No update. I cannot use my oneplus. I use Motorola only. Thanks for your consideration!
